### ISSUE TITLE
Getting bbox of element "g" is not possible

### ISSUE THREAD

**Issue by tmaziere** [svgdotjs/svgdom#73]
Hello,

First, thanks for such a useful tool!

I use this library to analyze SVGs which are -in other app- displayed and modified in the browser (with SVG.js v3, too).
For instance i would like to get the position of several elements (mostly `g`).
I set the default font through `config`, and then : 

```javascript
fs.readFile("./test.svg", "utf8", (err, data) => {
  if (err) {
    console.error(err);
    return;
  }
  const window = createHTMLWindow();
  const document = window.document;
  registerWindow(window, document);
  var svgDoc = SVG(document.documentElement);
  svgDoc.svg(data);
  var title = svgDoc.findOne("#my_g_element");
  console.log(title.x(), title.y());
});
```

It gives me an error with all tested elements: 

```
Error: Getting bbox of element "g" is not possible. TypeError: node.matrixify is not a function
    at getBox.call.el (/[...]/node_modules/@svgdotjs/svg.js/dist/svg.node.js:3124:13)
```

It should say that using `.x()` and `.y()` on these elements works well in the browser.

Is this an expected behavior?

**Comment by Fuzzyma** [comment]
I cant think of anything from the top of my head. Unrelated question: Is there a reason you create a HTMLWindow and not n SVGWindow?

To resolve the issue I would need the file you used so I can reproduce it on my end. Even better: You are able to narrow down the error to a simple case

**Comment by tmaziere** [comment]
No reason for the HTMLWindow, it was just a try.

I just created a simple SVG from scratch, 2 rects grouped in a `g` with id `layer1` (source below).

My complete example is now:

```javascript
const fs = require("fs");
const { createSVGWindow, config } = require("svgdom");
const { SVG, registerWindow } = require("@svgdotjs/svg.js");

config
  .setFontDir("./fonts")
  .setFontFamilyMappings({ Montserrat: "Montserrat-Regular.ttf" })
  .preloadFonts();

fs.readFile("./squares.svg", "utf8", (err, data) => {
  if (err) {
    console.error(err);
    return;
  }
  const window = createSVGWindow();
  const document = window.document;
  registerWindow(window, document);
  var svgDoc = SVG(document.documentElement)
  svgDoc.svg(data);
  var mygroup = svgDoc.findOne("#layer1");
  console.log(mygroup.svg());
  console.log(mygroup.cx(), mygroup.cy());
});
```

The first log returns the `g`element, that's OK.

But same error, then : 

`Getting bbox of element "g" is not possible. TypeError: node.matrixify is not a function`

Package versions:

```
"@svgdotjs/svg.js": "^3.0.16",
"svgdom": "^0.1.8"
```

And last, the `squares.svg` source: 

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg" width="297mm" height="210mm" viewBox="0 0 297 210" version="1.1" id="svg431822">
   <defs id="defs431816" />
   <metadata id="metadata431819">
      <rdf:RDF>
         <cc:Work rdf:about="">
            <dc:format>image/svg+xml</dc:format>
            <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
            <dc:title></dc:title>
         </cc:Work>
      </rdf:RDF>
   </metadata>
   <g id="layer1">
      <g id="g836">
         <rect style="fill:#ffffff;stroke:#080000;stroke-width:0.264583" id="rect830" width="33.261906" height="41.577381" x="67.279762" y="47.625" />
         <rect style="fill:#ffffff;stroke:#080000;stroke-width:0.264583" id="rect832" width="37.041668" height="44.601189" x="124.73214" y="46.113094" />
      </g>
   </g>
</svg>
```

**Comment by ItayBa** [comment]
Hey, I am having the same issue with some svgs , do anyone solved it?

**Comment by aloisklink** [comment]
I was having a similar issue. In my case, it was because my SVG file used a [`<foreignObject/>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject) SVG Element, which then had Text HTML elements.

So instead of using an [SVGTextElement](https://developer.mozilla.org/en-US/docs/Web/API/SVGTextElement), I instead had a plain HTML [Text](https://developer.mozilla.org/en-US/docs/Web/API/Text), which didn't have a `.matrixify()` method.

## Example

### Test Code

```js
const { createSVGWindow } = require('svgdom');
const window = createSVGWindow();
const document = window.document;
const { SVG, registerWindow } = require('@svgdotjs/svg.js');

// register window and document
registerWindow(window, document);

const svgString = require("fs").readFileSync("test.svg", "utf8");
const bbox = SVG(svgString).bbox();
```

### Test SVG

```svg
<svg xmlns="http://www.w3.org/2000/svg" id="mermaid-1636771878629" width="100%" height="100">
    <g/>
    <defs>
        <g id="def-commit">
            <circle r="10" cx="0" cy="0"/>
            <!-- !!!⚠️⚠️ Here: embedded XHTML tags in SVG ⚠️⚠️!!!!!!!! -->
            <foreignObject width="75" height="100" x="-25" y="0" class="node-label" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
                <p xmlns="http://www.w3.org/1999/xhtml"/>
            </foreignObject>
        </g>
    </defs>
    <g id="node-ed4eee2" class="commit" transform="translate(50, 50)" fill="yellow" stroke="grey" stroke-width="2">
        <circle r="10" cx="0" cy="0"/>
        <foreignObject width="75" height="100" x="-25" y="0" class="node-label" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
            <p xmlns="http://www.w3.org/1999/xhtml">
                <span class="branch-label">master, </span>
                <span class="commit-id">ed4eee2</span>
            </p>
        </foreignObject>
    </g>
</svg>
```

### Parsed SVG `console.log`

```js
> console.log(SVG(svgString));
Svg {
  events: {},
  node: SVGSVGElement {
    localName: 'svg',
    prefix: null,
    nodeName: 'svg',
    namespaceURI: 'http://www.w3.org/2000/svg',
    nodeType: 1,
    nodeValue: null,
    childNodes: [
      [Text], // this is where the Exception comes from
      [SVGGraphicsElement],
      [Text],
      [SVGGraphicsElement],
      [Text],
      [SVGGraphicsElement],
      [Text]
    ],
...
```

**Comment by Fuzzyma** [comment]
Sorry for the ultra late reply. I totally forgot about it. I just pushed a fix that should deal with that

**Comment by Fuzzyma** [comment]
Released with 0.1.9
